### ISSUE TITLE
Fix footnote issue in `prepare_ae_specific()`

### ISSUE THREAD

**Issue by nanxstats** [Merck/metalite.ae#4]
I tentatively fixed the R CMD check issue in `prepare_ae_specific()` by wrapping the `rtf_encode()` call inside a `try()` call.

Although the RTF output looks ok, the root cause is not fixed. I suspect this is related to how r2rtf handles vector inputs (specifically, footnotes), perhaps relevant to https://github.com/Merck/r2rtf/issues/88. Therefore, this might need a proper fix from upstream.

To reproduce the issue:

```r
library(metalite.ae)

meta <- meta_ae_dummy()

outdata <- meta |>
  prepare_ae_specific(
    population = "apat",
    observation = "wk12",
    parameter = "rel"
  ) |>
  format_ae_specific() |> tlf_ae_specific(
    footnotes = c("Footnote 1", "Footnote 2"),
    source = "Source:  [CDISCpilot: adam-adsl; adae]",
    medra_version = "24.0",
    path_outdata = "zzz.rda",
    path_outtable = "zzz.rtf"
  )
```

```
Warning messages:
1: In matrix(text_rtf, nrow = nrow(text), ncol = ncol(text)) :
  data length differs from size of matrix: [2 != 1 x 1]
2: In matrix(text_rtf, nrow = nrow(text), ncol = ncol(text)) :
  data length differs from size of matrix: [2 != 1 x 1]
3: In matrix(text_rtf, nrow = nrow(text), ncol = ncol(text)) :
  data length differs from size of matrix: [2 != 1 x 1]
4: In matrix(text_rtf, nrow = nrow(text), ncol = ncol(text)) :
  data length differs from size of matrix: [2 != 1 x 1]
```

The warning can be located to the footnote handling lines. The call chain is:

```
tlf_ae_specific -> rtf_output ->
rtf_encode -> rtf_encode_table -> as_rtf_footnote -> rtf_table_content -> rtf_text -> matrix
```

To reproduce this locally, one will need to use `R CMD check --as-cran` which takes warnings generated by code examples and vignettes seriously.

FYI @wangben718 @LittleBeannie

**Comment by elong0527** [comment]
Wow, the issue seems more complicated than I thought. I will check from my end.

**Comment by nanxstats** [comment]
I'm closing this as it is now fixed by r2rtf v0.3.5 + https://github.com/Merck/metalite.ae/commit/ccff57b4867b53b3d2cd3e0972343977fbdb3345 + https://github.com/Merck/metalite.ae/commit/ccbca87e7a5b29d91ccdadbf44c6e973be308005
